### PR TITLE
Atom 6815 handle for timeouts

### PR DESCRIPTION
**Pull request by dougperez**
## Ticket

https://jira.bqsoft.com/browse/ATOM-6815

## What this does

Handles for timeout

## Why we did this

Today we just blow up instead of handling timeouts like errors

https://developers.braintreepayments.com/reference/general/exceptions/ruby#timeout-exception

### CODE REVIEW

File: lib/conduit/braintree/actions/base.rb [hunk 1]
```diff
@@ -43,6 +43,8 @@ def report_braintree_exceptions(exception)
       when "Braintree::NotFoundError"
         identifier = @options[:token] || @options[:customer_id]
         respond_with_error("Failed to find resource with identifier #{identifier}")
+      when "Net::ReadTimeout"
```
Comment (pedromartinez) [inline lib/conduit/braintree/actions/base.rb]: Is this the class that gets returned?

Comment (pedromartinez) [reply lib/conduit/braintree/actions/base.rb]: I see these: https://github.com/braintree/braintree_ruby/blob/master/lib/braintree/exceptions.rb

Comment (pedromartinez) [reply lib/conduit/braintree/actions/base.rb]: Also see this issue: https://github.com/braintree/braintree_ruby/issues/88

Looks like we'll have to handle `Net::OpenTimeout` and `Errno::ETIMEDOUT`

File: lib/conduit/braintree/actions/base.rb [hunk 2]
```diff
@@ -18,12 +18,26 @@ def initialize(options)
     # Performs the request, with mocking if requested
     #
     def perform
+      body = perform_action
+      parser = parser_class.new(body)
+      Conduit::ApiResponse.new(raw_response: @raw_response, body: body, parser: parser)
+    rescue Conduit::NotFoundError
```
Comment (dougperez) [inline lib/conduit/braintree/actions/base.rb]: Anyone know what this is doing?

Comment (pedromartinez) [reply lib/conduit/braintree/actions/base.rb]: @dougperez It should be:

```ruby
    rescue Braintree::NotFoundError => error
      raise(Conduit::NotFoundError, error.message)
```

File: lib/conduit/braintree/actions/base.rb [hunk 3]
```diff
@@ -89,3 +103,5 @@ def mock_mode?
     end
   end
 end
+
+class Conduit::NotFoundError < StandardError; end
```
Comment (pedromartinez) [inline lib/conduit/braintree/actions/base.rb]: This seems like it's already defined.

**Comment by dougperez** [comment]
@pedromartinez Should be ready for another review.

**Pull request merged**